### ISSUE TITLE
Injecting trace headers?

### ISSUE THREAD

**Issue by obitech** [riandyrn/otelchi#4]
Hey, sorry in advance for the newbie question 🙈 

I've just been testing otelchi with some of our services but I think the trace headers are not injected on the outgoing request. Do I need to this manually? I couldn't find anything in the regarding injection in either this code base or the [otel docs](https://opentelemetry.io/docs/instrumentation/go/manual/).

Cheers!

**Comment by obitech** [comment]
Ok I've just realized I can use an instrumented [Transport](https://github.com/open-telemetry/opentelemetry-go-contrib/tree/instrumentation/net/http/otelhttp/v0.28.0/instrumentation/net/http/otelhttp)

**Comment by obitech** [comment]
@riandyrn sorry to bring this up again, but would it make sense to inject the trace ID into the outgoing request? otelchi is already wrapping the writer, why not write the trace ID into the outgoing header like in [otelhttp](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/net/http/otelhttp/handler.go#L168) ?

**Comment by riandyrn** [comment]
Hello, @obitech 

I don't think I understand the subject that you want to discuss. 😅

Are you perhaps asking why in the response writer in `otelchi` didn't also inject the tracing header?

![Untitled Diagram drawio (1)](https://user-images.githubusercontent.com/5761331/151509612-ff385f69-b8ba-485c-81c1-9b0c422ac842.png)

**Comment by obitech** [comment]
More like Service A is uninstrumented and sends a request to Service B. Shouldn't the response include the Trace ID?

**Comment by riandyrn** [comment]
Hmm.., ok, let us try to examine 2 possible scenarios of tracing:

1. `Service A` is not instrumented with `otelhttp` & `Service B` is instrumented with `otelchi`.
2. `Service A` is instrumented with `otelhttp` & `Service B` is instrumented with `otelchi`.

For the first scenario, why `Service B` need to put its `TraceID` in response header? I mean `TraceID` is originated inside `Service B`, so why `Service A` need to know about it? 😅

For the second scenario, what happens is like this:

1. `Service A` start tracing
2. `Service A` make http call to `Service B` via instrumented client using `otelhttp` passing the trace context
3. `otelhttp` inject trace headers into outgoing call
4. `Service B` handler that has been instrumented with `otelchi` receives the http call
5. `otelchi` will unpack the trace headers, put them in context, and pass these context to actual handler
6. Actual handler has been executed, response has been written to `Service A`
7. `otelchi` will generate tracing object and export it into collector (let say Jaeger)
8. `Service A` received the response from `Service B`
9. `otelhttp` will generate tracing object and export it into same collector as `Service B` (let say Jaeger)
10. Jaeger will try to correlate tracing object from both `Service A` & `Service B`
11. We could see the complete trace from `Service A` to `Service B` in Jaeger

Here is the diagram that shows overview of this process:

![what-is-distributed-tracing-3](https://user-images.githubusercontent.com/5761331/151560293-34be6df8-476f-45f7-b14f-21a8be22ccb4.png)

In short, injecting `TraceID` into final response of `otelchi` is not affecting context propagation in distributed tracing.

So what is your use case that make you think it is necessary to put `TraceID` in response header?

**Comment by riandyrn** [comment]
Hello, @obitech

Is there any update for this issue?

My main motivation to continue the discussion is maybe you can see use cases that I couldn't see. 😃

**Comment by obitech** [comment]
Hey, sorry I was a bit caught up the past two weeks. I appreciate your diagram and explanation, it does make sense to me. I just don't understand why the otelhttp Handler is [injecting the trace ID](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/net/http/otelhttp/wrap.go#L94) on the [response out](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/net/http/otelhttp/handler.go#L168), do you have any idea?

**Comment by riandyrn** [comment]
> do you have any idea?

Currently no. If you later found out the reason, let us know. 😃
